Source/Fields/LMCCylindricalCavity.cc

```diff
@@ -77,25 +77,33 @@ namespace locust
     	return tIntegral;
     }
 
-    double CylindricalCavity::Z_TE(int l, int m, int n) const
+    double CylindricalCavity::Z_TE(int l, int m, int n, double fcyc) const
     {
+    	double Z_TE = 1.0;
     	double x_lm = fInterface->fBesselNKPrimeZeros[l][m];
     	double k1 = x_lm / fInterface->fR;
     	double k3 = n * LMCConst::Pi() / fInterface->fL;
     	double k = pow(k1*k1+k3*k3,0.5);
-    	double eta = sqrt( LMCConst::MuNull() / LMCConst::EpsNull() );  // Pozar p. 291.
-    	double Z_TE = k * eta / k3; // Pozar p. 286, Jackson Eq. 8.32
+    	double k0 = fcyc / LMCConst::C();
+    	if ( k*k-k0*k0 != 0. )
+    	{
+    		Z_TE *= fcyc/(k0*k0 - k*k);  // after Collin Foundations of M.E. Eq. 7.132
+    	}
```

Comment:
I know this is still a work in progress, but do these impedance factors have other components that will be incorporated later? Perhaps I'm misreading the referenced equation but it seems like there are more components to the given term.

Reply:
Thanks and yes, it does need a closer look in the context of the other steps in the simulation.  So this first step only contains the frequency and wavenumbers as a first pass.